src/model.py

```
@@ -489,15 +574,42 @@ def forward(self, idx):
+        args = self.args
         idx, targets = batch
-        logits = self(idx)
-        loss = F.cross_entropy(
-            logits.view(-1, logits.size(-1)), targets.view(-1))
-        return L2Wrap.apply(loss, logits)
+        
+        if args.fuse_l2warp:
+            # Use fused linear cross entropy with L2Warp
+            # This avoids materializing the full logits tensor
+            hidden = self(idx, return_logits=False)  # [B, T, n_embd]
+            B, T, H = hidden.shape
```

Ruff (0.15.6)

[warning] Unpacked variable `B` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)